Multilayer

fnobregasantos opened the pull request:
Review Multialyer - Networks code

fnobregasantos committed:
Import main multilayer code

fnobregasantos committed:
Delete Multilayer

fnobregasantos committed:
Created a readme file to guide the users

fnobregasantos committed:
Main codes and a training matrix with MST 

The matrixes are analogous to the ones created in the lab, for ease our use, just that, for privacy reasons,  all values are random

fnobregasantos committed:
Included docstrings of some functions

fnobregasantos committed:
Included missing docstrings

fnobregasantos committed:
Included degree centrality

fnobregasantos committed:
Included clustering

fnobregasantos committed:
Proofread the code (checking for typos, etc.)

fnobregasantos committed:
Fixing some typos

fnobregasantos committed:
Fixing some typos

multinetlab committed:
Merge branch 'master' into Multilayer

eduardacenteno committed:
Rephrasing

eduardacenteno commented on Multilayer/readme.txt:
```diff
@@ -0,0 +1 @@
+This folder includes all scripts for the Multilayer analysis . All instructions will be found here!
```
I would recommend, at some point, writing in more detail a tutorial explaining these analyses/scripts.

eduardacenteno committed:
Correcting style // typos

I corrected a few typos but mainly changed the style following PEP8 suggestions

fnobregasantos committed:
Delete Multilayer_Main_code.py

It was in wrong place - sorry!

fnobregasantos committed:
Improved code - Now only with main functions

fnobregasantos committed:
Preliminary version of the Multilayer_user code

Build upon the Main Multilayer code

lcbreedt committed:
lucas review and readme of fernando's code

lcbreedt committed:
lucas/fernando review and revision of the code

lcbreedt committed:
lucas/fernando review and revision of the code

lcbreedt committed:
lucas revision of readme.md

lcbreedt committed:
lucas revision of readme.md - fixed ToC

lcbreedt committed:
lucas revision of readme.md - fixed ToC

lcbreedt committed:
lucas revision of readme.md - fixed ToC + added sections

lcbreedt committed:
lucas revision of readme.md - fixed ToC + added sections

fnobregasantos committed:
fixed typo in dataframe

fnobregasantos committed:
The code also reads 2d arrays (not only 3d arrays)

The original code uses as input a 3d array, i.e., two variables for the shape of the supra-adjacency matrix and one variable for the number of matrixes. If we have a single supra-adjacency matrix, we often represent it as a 2d array, which would yield a bug in our code.  Although the original code was not wrong, we decided to facilitate the user experience by adapting the code to accept a single supra-adjacency matrix as a 2d array as input. When that happens, the number of individuals in the analysis will be assigned one in the code.

fnobregasantos committed:
The code also accepts 2d arrays as input

The original code uses as input a 3d array, i.e., two variables for the shape of the supra-adjacency matrix and one variable for the number of matrixes. If we have a single supra-adjacency matrix, we often represent it as a 2d array, which would yield a bug in our code.  Although the original code was not wrong, we decided to facilitate the user experience by adapting the code to accept a single supra-adjacency matrix as a 2d array as input. When that happens, the number of individuals in the analysis will be assigned one in the code.

multinetlab commented:
@fnobregasantos @lcbreedt Before your next commit, please have a look at the files in the repo and check which should be kept. There's one code outside the Multilayer folder, and another one inside.

fnobregasantos committed:
Comment 1 

We commented that you should always start with 0 as the first layer, not 1 (like in other programs such as Matlab).

fnobregasantos committed:
Comment 2

Included example to path similar to the one described by Marika

fnobregasantos committed:
Comment 3

Including a reminder to subtract -1 from your subnetwork list

fnobregasantos committed:
Comment 4

Reminder for a path to output file

fnobregasantos committed:
Revert old commit

This code was a user file before the previous commit. It was wrongly overwritten with the Multilayer main code. This commit is only reverting it.
